cupy/sorting/search.py

```diff
@@ -24,7 +24,27 @@ def argmax(a, axis=None, dtype=None, out=None, keepdims=False):
     return a.argmax(axis=axis, dtype=dtype, out=out, keepdims=keepdims)
 
 
-# TODO(okuta): Implement nanargmax
+def nanargmax(a, axis=None, dtype=None, out=None, keepdims=False):
+    """Return the indices of the maximum values in the specified axis ignoring
+    NaNs. For all-NaN slice ``ValueError`` is raised.
```

Comment:
```suggestion
    NaNs. For all-NaN slice ``-1`` is returned.
```

Reply:
Bug:
```py
>>> import cupy
>>> cupy.nanargmin(cupy.array([float('nan')]))
array(0, dtype=int64)
>>> cupy.nanargmin(cupy.array([float('nan'), float('nan')]))
array(-1, dtype=int64)
```
`cupy.nanargmin(cupy.array([float('nan')]))` should be return `-1`.

Reply:
The code simplification I posted [here](https://github.com/cupy/cupy/pull/2222#issuecomment-510323287) will fix the above bug.

Reply:
Got it, Thanks. I will be more helpful if there was documentation on what's going on those line with reduction function. I have made all the changes now.

Reply:
`min_max_st<type_in0_raw>` has two member variables: the number of index `_J` and `type_in_raw0` value `in0`. This fix assigns `-1` instead of `_J` as an index when `in0` is NaN.

Reply:
Alright got it. This simplification fixes all the changes I did previously and renders them unnecessary I suppose. i.e my_nanargmin and my_nanargmin_float, etc